lm_eval/api/task.py

```diff
@@ -51,13 +51,63 @@
 eval_logger = logging.getLogger("lm-eval")
 
 
+@dataclass
+class GroupConfig(dict):
+    group: Optional[Union[str, list]] = None
+    aggregate_metric: Optional[str] = False
+    aggregate_fn: Optional[str] = "mean"
```

Comment:
aggregate fn being defined explicitly is good, I think we also need to try to be careful of how tasks with (multiple) *filters* interact with groups. 

Currently, if two tasks have filters with different names, a group can't aggregate those correctly. We may want to have groups, for each metric they'll report, expressly define, for each subtask, what metric and what filter to include in the aggregation.